source/common/http/conn_manager_utility.cc

```diff
@@ -92,8 +92,30 @@ ConnectionManagerUtility::MutateRequestHeadersResult ConnectionManagerUtility::m
   if (!Utility::isUpgrade(request_headers)) {
     request_headers.removeConnection();
     request_headers.removeUpgrade();
+
     if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.sanitize_te")) {
-      request_headers.removeTE();
+      auto teHeader = request_headers.getTEValue();
```

Comment:
Also these local variable names should be `snake_case` not `mixedCase`.

https://google.github.io/styleguide/cppguide.html#Variable_Names